articles/zk-snarks-under-the-hood/zk_snarks_under_the_hood.md

```diff
@@ -0,0 +1,147 @@
+
+For example, instead of sending $a*G + b*G*t + c*G*t^2$, the prover will send a point $a'*G$ instead.
+
+Our coefficients in the linear combination at the end of the QAP step were $1, x, x^2, x^3, ... , x^m$ but we are going to evaluate our polynomials at t, so they will turn into $G, G*t, G*t^2, G*t^3, ..., G*t^m$.
+
+Hence, so far the prover must give:
+
+- $π_a = G * A(t), π’_a = G * A(t) * k_a$
+- $π_b = G * B(t), π’_b = G * B(t) * k_b$
+- $π_c = G * C(t), π’_c = G * C(t) * k_c$
+
+Where $A(t)$ is a linear combination of $A_i$ evaluated at t and so on.
+
+Note that the prover doesn’t actually need to know (and shouldn’t know!) $t, k_a, k_b$, or $k_c$ to compute these values; rather, the prover should be able to compute these values just from the points that we’re adding to the [trusted setup](../../terms/trusted_setup.md).
+
+![Verification algorithm - all steps](./attachments/verification_steps.png)
+
+To evaluate the proof, we have to check 3 things:
+
+- Validity of knowledge commitments for A, B, C or check if the prover used the correct $k_a, k_b, k_c$
+- If the same coefficients $G, G_t, G_t^2, ...$ were used in the linear combination $\pi_a, \pi_b, \pi_c$
+- Finally, check if the equation is correct like in the QAP step
+
+Note that equation checking is done in [pairing](../../terms/elliptic_curve.md#elliptic-curve-pairings) because we have secret numbers k, t and [elliptic curve](../../terms/elliptic_curve.md) involved, unlike in the QAP step where x1, x2, x3, x4 are predetermined.
+# QAP Divisibility
+
+We need to prove that A * B - C = H * Z. We do this once again with a [pairing](../../terms/elliptic_curve.md#elliptic-curve-pairings) check:
+
+$e(\pi_a,\pi_b) / e(\pi_c, G) \stackrel{?}{=} e(\pi_h, G * Z(t))$
+
+Where $\pi_h=G * H(t)$ and $H$ is the remainder polynomial. If the connection between this equation and A * B - C = H * Z does not make sense to you, go back and read [our file on pairing](../../terms/elliptic_curve.md#elliptic-curve-pairings). Note that $e(\pi_c, G)$ = $\pi_c$ but we do this in order to use pairing math.
+
+We saw above how to convert A, B, and C into elliptic curve points; G is just the generator (i.e., the elliptic curve point equivalent of the number one). We can add $G * Z(t)$ to the trusted setup. H is harder; H is just a polynomial, and we predict very little ahead of time about what its coefficients will be for each individual QAP solution. Hence, we need to add yet more data to the trusted setup; specifically, the sequence:
```

Comment:
Please note that this file serves as a very concise version of the original blog. It should contains only key ideas/concepts that you find everyone should know about.

Reply:
So it will basically be a tl:dr with some explaination, right?